### PR TITLE
netdev_driver_t::recv(): Clarified documentation

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description
This PR clarifies the documentation in `netdev_driver_t::recv()` a bit and brings it back in sync with the code.

Changes in detail:

 - Precondition `(buf != NULL) && (len > 0)` removed, because it is strictly incorrect
 - Changed `@brief` text to reflect the the three different features of `netdev_driver_t::recv()`. (Leading with the least obvious one, to attract more attention to it.)
 - Clarified how to behave correctly when receiving a frame larger than the supplied buffer
 - Clarified that `recv()` with `(buf == NULL) && (len == 0)` may also return an upper bound estimation of the frame size, instead of the actual frame size.

### Testing procedure
Doesn't apply.

### Issues/PRs references
- Resolves https://github.com/RIOT-OS/RIOT/issues/10413
- May also help to keep bugs like in https://github.com/RIOT-OS/RIOT/issues/10410 from re-appearing

### CODE REVIEW

File: drivers/include/net/netdev.h [hunk 1]
```diff
@@ -310,19 +310,26 @@ typedef struct netdev_driver {
     int (*send)(netdev_t *dev, const iolist_t *iolist);
 
     /**
-     * @brief Get a received frame
+     * @brief Drop a received frame, *OR* get the length of a received frame,
```
Comment (miri64) [inline drivers/include/net/netdev.h]: Maybe even make the **OR** bold?
```suggestion
      * @brief Drop a received frame, **OR** get the length of a received frame,
```

**Comment by miri64** [comment]
Allowed myself to change the title ;-). Also commits needs typo fixes as well. BTW with some fixes to make the "Clarified documenation" summaries more specific, I don't think we need to squash the commits.

**Comment by maribu** [comment]
Ok, I tried to come up with better commit messages. I changed the prefix "drivers/netdev_driver_t:" to "netdev_driver_t::recv:", in order to keep the first line in the character limit. I also applied your comment regarding the bold OR. The result can be seen here: https://github.com/maribu/RIOT/commits/netdev_driver_doc. Should I force push?

I think my spell checker misses the typos and I was unable to find them manually. Do you mind to point them out?

**Comment by miri64** [comment]
Yes please force push

> I think my spell checker misses the typos and I was unable to find them manually. Do you mind to point them out?

I was specifically referring to this commit message:

> drivers/netdev_driver_t: Clarified docume**na**tion

**Comment by miri64** [comment]
Maybe not mention the user `@pre` in the commit message ;-) he might get annoyed when people do stuff to their GitHub fork ;-)

**Comment by miri64** [comment]
(same goes for `@brief`)

**Comment by miri64** [comment]
Other than that: my ACK holds, but needs more than one.

**Comment by maribu** [comment]
I removed the `@`before `@brief section` (as it remains clear without `@`) and re-phrase the `@pre` commit header here: https://github.com/maribu/RIOT/commits/netdev_driver_doc Force push?

**Comment by miri64** [comment]
Yepp.

**Comment by miri64** [comment]
Needs >1 ACK. Just so people know why we not merge! ;-)

**Comment by maribu** [comment]
@gebart: I do agree that using correct terms (frame for layer 2, packet for layer three, datagram/segment for layer 4) would be much better. But maybe we should keep the term packet for now here to use it consistently. In a separate PR we could fix the usage of terms.

**Comment by gebart** [comment]
@maribu I agree, it is not the goal of this PR.

**Comment by miri64** [comment]
Agreed as well. However, the "it" wording pointed out by @gebart should be fixed. You can squash immediately.

**Pull request merged**